descriptions/api.github.com/api.github.com.yaml

```diff
@@ -19753,6 +19887,204 @@ paths:
         previews: []
         category: repos
         subcategory: 
+  "/repos/{owner}/{repo}/environments":
+    get:
+      summary: Get all environments
+      description: |-
+        Get all environments for a repository.
+
+        Anyone with read access to the repository can use this endpoint. If the repository is private, you must use an access token with the `repo` scope. GitHub Apps must have the `actions:read` permission to use this endpoint.
+      tags:
+      - repos
+      operationId: repos/get-all-environments
+      externalDocs:
+        description: API method documentation
+        url: https://docs.github.com/rest/reference/repos#get-all-environments
+      parameters:
+      - "$ref": "#/components/parameters/owner"
+      - "$ref": "#/components/parameters/repo"
+      responses:
+        '200':
+          description: response
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  total_count:
+                    description: The number of environments in this repository
+                    example: 5
+                    type: integer
+                  environments:
+                    type: array
+                    items:
+                      "$ref": "#/components/schemas/environment"
+              examples:
+                default:
+                  "$ref": "#/components/examples/environments"
+      x-github:
+        githubCloudOnly: false
+        enabledForGitHubApps: true
+        previews: []
+        category: repos
+        subcategory: environments
+  "/repos/{owner}/{repo}/environments/{environment_name}":
+    get:
+      summary: Get an environment
+      description: Anyone with read access to the repository can use this endpoint.
+        If the repository is private, you must use an access token with the `repo`
+        scope. GitHub Apps must have the `actions:read` permission to use this endpoint.
+      tags:
+      - repos
+      operationId: repos/get-environment
+      externalDocs:
+        description: API method documentation
+        url: https://docs.github.com/rest/reference/repos#get-an-environment
+      parameters:
+      - "$ref": "#/components/parameters/owner"
+      - "$ref": "#/components/parameters/repo"
+      - "$ref": "#/components/parameters/environment_name"
+      responses:
+        '200':
+          description: response
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/environment"
+              examples:
+                default:
+                  "$ref": "#/components/examples/environment"
+      x-github:
+        githubCloudOnly: false
+        enabledForGitHubApps: true
+        previews: []
+        category: repos
+        subcategory: environments
+    post:
```

Comment:
@xuorig @imwiss You probably had an internal discussion about this and there is a good reason, but just to be sure: do we need the `POST /repos/{owner}/{repo}/environments/{environment_name}` if there already is `PUT /repos/{owner}/{repo}/environments/{environment_name}`? I think this is incoherent with the rest of the API. WE have several endpoints that create or update using the `PUT` method, why didn't you do the same here?

Reply:
@gr2m we just made changes to this API (in beta). It is now consistent and using PUT for both creation and update. Thanks for catching this early 🙇

Reply:
The update to the spec is coming.

Reply:
Thank you, got it 👍🏼